src/test/controllers/UserNewsController.test.js

```diff
@@ -0,0 +1,212 @@
+const app = require("../../../index");
+const request = require("supertest");
+
+let token = "";
+let userId = "";
+
+beforeAll((done) => {
+  const user = {
+    fullName: "Jane Smith",
+    email: "janesmith123@mail.com",
+    password: "jane1234",
+  };
+  request(app)
+    .post("/signup")
+    .send(user)
+    .end((err, res) => {
+      token = res.body.data.accessToken;
+      userId = res.body.data.user.id;
+      if (userId) {
+        request(app)
+          .put(`/users/${userId}/news/preferences`)
+          .set("authorization", token)
+          .send({
+            sources: "bbc-news",
+            country: "in",
+          })
+          .end((err, res) => {
+            done();
+          });
+      }
+    });
+});
+
+describe("Test cases of fetch user's news preferences API", () => {
+  test("User's news preferences have been fetched successfully", (done) => {
+    request(app)
+      .get(`/users/${userId}/news/preferences`)
+      .set({ Authorization: token })
+      .end((err, res) => {
+        expect(res.statusCode).toBe(200);
+        expect(res.body.status).toBeTruthy();
+        expect(res.body.message).toMatch(
+          "News preference have been fetched successfully."
+        );
+        done();
+      });
+  });
+
+  test("User with the provided id doesn't exists", (done) => {
+    request(app)
+      .get(`/users/23132/news/preferences`)
+      .set({ Authorization: token })
+      .end((err, res) => {
+        expect(res.statusCode).toBe(404);
+        expect(res.body.status).toBeFalsy();
+        expect(res.body.message).toMatch(
+          "User with the provided id doesn't not exists."
+        );
+        done();
+      });
+  });
+
+  test("User is unauthorized", (done) => {
```

Comment:
instead of doing this for each API call, just write a unit test for the auth middleware